src/mixed.js

```diff
@@ -108,7 +108,7 @@ SchemaType.prototype = {
     let path = state.path
 
     let errors = [];
-    let reject = () => Promise.reject(new ValidationError(errors, value));
+    let reject = () => Promise.reject(new ValidationError(errors, value, path));
```

Comment:
`errors` here may be more than just the type error, it could also contain the whitelist and black list errors. This works fine when It's just a type error since that is the first that can be thrown (hence only one), but will break when its used in either the blacklist or whitelist code path.

The main "bug" here is that ValidationError creates an aggregate Error when passed an array of ValidationErrors, but not when it is passed an error of error strings